BatchNormalization.lua

```diff
@@ -154,3 +154,16 @@ function BN:accGradParameters(input, gradOutput, scale)
       self.gradBias:add(scale, self.buffer)
    end
 end
+
+function BN:clearState()
+   for i,v in ipairs{
+      'buffer',
+      'buffer2',
+      'centered',
+      'std',
+      'normalized',
+   } do
+      if self[v] then self[v]:set() end
+   end
```

Comment:
I had this at some point: https://gist.github.com/szagoruyko/bd39620cdf35964647b2 what do you think?